### PR TITLE
DAO-196 Sanitise URL in ExternalLink component

### PR DESCRIPTION
**Pull request by mcoetzee**
### What does this change?
Updates the ExternalLink component to only accept URLs that start with "https://" or "http://"

### CODE REVIEW

File: src/components/external-link/external-link.tsx [hunk 1]
```diff
@@ -9,7 +9,13 @@ interface Props {
 }
 
 const ExternalLink = (props: Props) => {
-  const { className, href, children } = props;
+  const { className, children } = props;
+
+  let href = props.href.trim();
+  const urlRegex = /^https:\/\//i; // Starts with https:// (case insensitive)
```
Comment (Siegrift) [inline src/components/external-link/external-link.tsx:15]: Mhh, we already do this precaution in https://github.com/api3dao/api3-dao-dashboard/blob/main/src/pages/proposal-commons/proposal-details/proposal-details.tsx#L39 but we also accept `http`

Any reason why only accept https?

Comment (mcoetzee) [reply src/components/external-link/external-link.tsx:15]: Just for some more context, this PR is more of a precautionary change (I created a proposal with a dodgy javascript URL and saw that it did not create a link for it). 

W.r.t. `http`, ah I missed the fact that we already allow them, so I will make the change to bring them back.

Comment (Siegrift) [reply src/components/external-link/external-link.tsx:15]: Also, it would be nice to show a warning (at least to console). WDYT?

Comment (mcoetzee) [reply src/components/external-link/external-link.tsx:15]: Sounds good 👍

**Review by Siegrift — APPROVED** [review]
👍 LGTM

**Pull request merged**